packages/react/src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useLayoutEffect, useRef, createRef } from 'react';
+import PropTypes from 'prop-types';
+import root from 'window-or-global';
+import { ArrowRight20, ArrowDown20, Pdf20 } from '@carbon/icons-react';
+import { Button } from 'carbon-components-react';
+import { settings } from 'carbon-components';
+import { BUTTON_GROUP } from '../../internal/FeatureFlags';
+import {
+  featureFlag,
+  settings as ddsSettings,
+} from '@carbon/ibmdotcom-utilities';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * renders the buttons (max 2 buttons)
+ *
+ * @param {object} props props object
+ * @param {Array} props.buttons array of buttons for lead space (max 2 buttons)
+ * @returns {*} button components
+ */
+const ButtonGroup = ({ buttons }) => {
+  const icons = {
+    ArrowDown: ArrowDown20,
+    ArrowRight: ArrowRight20,
+    Pdf: Pdf20,
+  };
+
+  const buttonsRef = useRef(buttons.map(() => createRef()));
+
+  /**
+   * ensure buttons have equal width, based off the
+   * largest width of the two
+   */
+  function adjustWidths() {
+    const button1 = buttonsRef.current[0];
+    const button2 = buttonsRef.current[1];
+
+    if (button1.current.offsetWidth !== button2.current.offsetWidth) {
+      button1.current.offsetWidth > button2.current.offsetWidth
+        ? (button2.current.style.width = `${button1.current.offsetWidth}px`)
+        : (button1.current.style.width = `${button2.current.offsetWidth}px`);
+    }
+  }
+
+  useLayoutEffect(() => {
+    if (buttonsRef.current.length > 1) {
+      root.addEventListener('load', adjustWidths);
+      root.addEventListener('resize', adjustWidths);
+    }
+    return () => {
+      root.removeEventListener('load', adjustWidths);
+      root.removeEventListener('resize', adjustWidths);
+    };
+  }, []);
+
+  return featureFlag(
+    BUTTON_GROUP,
+    <div
+      className={`${prefix}--buttongroup`}
+      data-autoid={`${stablePrefix}--buttongroup`}>
+      <div className={`${prefix}--buttongroup__ctas`}>
+        {buttons.map((button, key) => {
+          if (key > 1) return;
```

Comment:
Does it make sense to pull this `1` up and give it a const value to help define what's happening here a little better?

```javascript
const BUTTON_COUNT_ALLOWED = 1; // count starts with 0

... 

if (key > BUTTON_COUNT_ALLOWED) return;

...

Reply:
Bringing @wonilsuhibm @larahanlon2 @oliviaflory here, and cc: @annawen1. Is there a requirement to cap the number of allowed buttons? I'm wondering if we should remove entirely?

cc: @photodow

Reply:
when I initially wrote this up as part of the `<LeadSpace />` component, there was a max of 2 buttons we could have for leadspace specifically. But since we're making the buttons more generic we should probably remove that.

Reply:
@annawen1 it could be something you set in the `Leadspace` itself where you only pass in the first two buttons in the array into the `ButtonGroup`.

Reply:
@photodow @annawen1 I removed the cap, and also I removed the JS code for adjusting widths and replaced with a pure flexbox solution.
cc: @wonilsuhibm @larahanlon2 @oliviaflory